pkg/kubelet/container/helpers.go

```diff
@@ -110,28 +110,20 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 // Note: remember to update hashValues in container_hash_test.go as well.
 func HashContainer(container *v1.Container) uint64 {
 	hash := fnv.New32a()
-	// Omit nil or empty field when calculating hash value
-	// Please see https://github.com/kubernetes/kubernetes/issues/53644
-	containerJSON, _ := json.Marshal(container)
+	containerJSON, _ := json.Marshal(pickFieldToHash(container))
 	hashutil.DeepHashObject(hash, containerJSON)
 	return uint64(hash.Sum32())
 }
 
-// HashContainerWithoutResources returns the hash of the container with Resources field zero'd out.
-func HashContainerWithoutResources(container *v1.Container) uint64 {
-	// InPlacePodVerticalScaling enables mutable Resources field.
-	// Changes to this field may not require container restart depending on policy.
-	// Compute hash over fields besides the Resources field
-	// NOTE: This is needed during alpha and beta so that containers using Resources but
-	//       not subject to In-place resize are not unexpectedly restarted when
-	//       InPlacePodVerticalScaling feature-gate is toggled.
-	//TODO(vinaykul,InPlacePodVerticalScaling): Remove this in GA+1 and make HashContainerWithoutResources to become Hash.
-	hashWithoutResources := fnv.New32a()
-	containerCopy := container.DeepCopy()
-	containerCopy.Resources = v1.ResourceRequirements{}
-	containerJSON, _ := json.Marshal(containerCopy)
-	hashutil.DeepHashObject(hashWithoutResources, containerJSON)
-	return uint64(hashWithoutResources.Sum32())
+// pickFieldToHash pick fields that will affect the running status of the container for hash,
+// currently this field range only contains `image` and `name`.
+// Note: this list must be updated if ever kubelet wants to allow mutations to other fields.
+func pickFieldToHash(container *v1.Container) map[string]string {
+	retval := map[string]string{
+		"name":  container.Name,
```

Comment:
why not include fields like ID, or ImageID? These should also be immutable. Are you finding the smallest set of fields that don't require a restart?

Reply:
This is picking the smallest set of mutable fields in the REST API v1.Container that *do* require a restart.

ID and ImageID are not in the v1.Container type

Reply:
> why not include fields like ID, or ImageID?


We already had some discussion above about not include the pod ID.
ref: https://github.com/kubernetes/kubernetes/pull/124220#discussion_r1577037684 https://github.com/kubernetes/kubernetes/pull/124220#discussion_r1559888960



> Are you finding the smallest set of fields that don't require a restart?

Yes, we would like your help in reviewing and verifying whether other fields in `v1.Container` need to be covered.

Reply:
gotcha, thanks!